<a href="https://colab.research.google.com/github/jldeato/UM-Flint-Coursework/blob/main/Deaton_HW1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Use pandas to read the olympic_medals csv

In [ ]:
# import pandas
import pandas as pd

# read the csv file into data frame
medals = pd.read_csv("/content/drive/MyDrive/DSC502_DATA/olympic_medals.csv")



Find the size of the data frame. How many rows? How many columns?

In [ ]:
# view info about the data frame
medals.shape

(2394, 8)

There are 2394 rows and 8 columns in the data frame.

Use the appropriate function to report the data types for all columns.

In [ ]:
# view information about the data frame
medals.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2394 entries, 0 to 2393
Data columns (total 8 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   Gender       2394 non-null   object
 1   Event        2394 non-null   object
 2   Location     2394 non-null   object
 3   Year         2394 non-null   int64 
 4   Medal        2394 non-null   object
 5   Name         2164 non-null   object
 6   Nationality  2394 non-null   object
 7   Result       2351 non-null   object
dtypes: int64(1), object(7)
memory usage: 149.8+ KB


Find how many cities matches were held at.

In [ ]:
# view number of unique locations
len(medals["Location"].unique())

23

Find how many medals the USA team won

In [ ]:
# count medals won by usa, isolate by nationality and medal column
medals[medals["Nationality"] == "USA"]["Medal"].count()


np.int64(638)

Find the total number of medals for each Nationality.

In [ ]:
# count medals for each nationality
medals.groupby(by="Nationality")["Medal"].count()

Nationality
ALG      9
ARG      4
AUS     57
AUT      4
BAH     12
      ... 
URS    131
USA    638
VEN      2
YUG      1
ZAM      1
Name: Medal, Length: 97, dtype: int64